MPMap.html

```diff
@@ -0,0 +1,241 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8" />
+    <title>My Climbs</title>
+    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
+        integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY="
+        crossorigin=""/>
+     <script src="https://unpkg.com/leaflet@1.9.4/dist/leaflet.js"
+        integrity="sha256-20nQCchB9co0qIjJZRGuk2/Z9VM+kNiyxNV1lvTlZBo="
+        crossorigin=""></script>
+        <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script> 
+
+
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <style>#map { height: 800px; width: 800px }</style>
+</head>
+<body>
+    <h3>Enter your mountain Project url</h5>
+    <input type="text" id="mpUser">
+    <button onclick="getProfile()">Submit</button>
+    <br><br>
+    <div id="map"></div>
+
+    <script>
+        let routeUrls = [];
+        let areaUrls = [];
+        let currentLat;
+        let currentLong;
+        let routeNameArr = [];
+        let gpsCoords = [];
+        let mark;
+        let markArray = [];
+
+        // Climb grades (5.9, WI4, etc.)
+        let rockGrade;
+        let iceGrade;
+        let mixedGrade;
+        let aidGrade;
+        let aidRating = [];
+
+        for(let k = 0; k < 6; k++){
+            aidRating.push("A" + k, "C" + k);
+        }
+
+        // Layers for markers and tooltips of climbs 
+        let rockLayer = L.layerGroup();
+        let iceLayer = L.layerGroup();
+        let mixedLayer = L.layerGroup();
+        let aidLayer = L.layerGroup();
+
+        // Arrays to hold grades of climbs
+        let rockDifficulty = [];
+        let iceDifficulty = [];
+        let mixedDifficulty = [];
+        let aidDifficulty = [];
+
+        // Temporary string to operate on if statement
+        let iceTemp;
+
+        // Set initial map view position
+        let map = L.map('map').setView([49.545206,-122.629395], 9);
+
+        // Terrain map layer
+        let OpenTopoMap = L.tileLayer('https://{s}.tile.opentopomap.org/{z}/{x}/{y}.png', {
+            maxZoom: 17,
+            attribution: 'Map data: &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
+        });
+        OpenTopoMap.addTo(map);
+
+        //CORS header for querying, get access by following https://cors-anywhere.herokuapp.com/corsdemo
+        jQuery.ajaxPrefilter(function(options) {
+            if (options.crossDomain && jQuery.support.cors) {
+                options.url = 'https://cors-anywhere.herokuapp.com/' + options.url;
+            }
+        });
+        function getProfile() {
```

Comment:
`getProfile` is turning into what is sometimes known as a "god" function because it's all seeing and all doing. I would suggest breaking `getProfile` down into more cohesive chunks of logic/functions.

We could start by breaking out the map logic from the `getProfile()` logic into a separate function. Something like:

```
let routesToDo = getProfileRoutes();
addRoutesToMap(routesToDo);
```

Then your `addRoutesToMap` function can handle the mapping layers/logic separately. When it's structured this way, you can then re-use `getProfileRoutes()` in other things like if you wanted to use the data to also display some text or a table below the map as an example.